crates/core-subsystem/core-resolver/src/validation/arguments_validator.rs

```diff
@@ -432,7 +432,7 @@ fn get_schema_directives(schema: &Schema, base_type_name: &str) -> Vec<ConstDire
             td.fields()
                 .unwrap_or(&placeholder)
                 .iter()
-                .filter(|x| x.node.ty.node.base.to_string() == base_type_name)
+                .find(|x| x.node.name.node.as_str() == argument_name)
```

Comment:
Perfect! Indeed, a field name is unique within a type (we check for it and issue an error if that is not the case).